packages/legend-query/src/stores/QueryBuilderState.ts

```diff
@@ -298,6 +316,18 @@ export class QueryBuilderState {
     );
   }
 
+  buildQueryParametersForUnsupportedState(): RawLambda {
```

Comment:
I keep looking for where we assert the `query` we support must be a `LambdaFunction`, turns out it's on line 291. Could you modify the `guaranteeType(...)` check there to throw a meaningful error message? Something like `Can't build query state: query builder only support lambda`